### PR TITLE
Implement inputstream.skip

### PR DESCRIPTION
**Pull request by mburman**

### CODE REVIEW

File: java/com/facebook/crypto/streams/NativeGCMCipherInputStream.java [hunk 1]
```diff
@@ -136,6 +136,6 @@ public synchronized void reset() throws IOException {
 
   @Override
   public long skip(long byteCount) throws IOException {
-    throw new UnsupportedOperationException();
+    return super.skip(byteCount);
```
Comment (siyengar) [inline java/com/facebook/crypto/streams/NativeGCMCipherInputStream.java]: wouldn't we still need to process the bytes, otherwise the mac wouldn't match

Comment (helios175) [inline java/com/facebook/crypto/streams/NativeGCMCipherInputStream.java]: I think the same. It needs to do exactly what read() does except maybe for copying the decrypted chunks into the final buffer...

Comment (helios175) [inline java/com/facebook/crypto/streams/NativeGCMCipherInputStream.java]: Yes. Thing is super.skip (InputStream.skip) already does that. It creates a byte[2048] an reads on it until end of stream, or byteCount. So it works.

A small improvement could be:
- having our own uniquely instantiated buffer (per stream, so repeated calls don't instantiate more buffers)
- implement read(long) as one of the loops in read *without* the call to System.arraycopy. Maxing the byte count in the buffer size. That way the loop is implemented naturally by the client calling more times to skip when needed (skip, as read, can return less bytes skipped).

But this solution works.

Comment (siyengar) [inline java/com/facebook/crypto/streams/NativeGCMCipherInputStream.java]: i would rather avoid creation of the 2048 byte buffer if possible to be as memory efficient as we can.

Comment (helios175) [inline java/com/facebook/crypto/streams/NativeGCMCipherInputStream.java]: So you'd go for the custom implementation of skip that does strictly what's needed? Sounds good.

**Comment by helios175** [comment]
Closing in favor of #121.

**Pull request closed without merging**